tests/src/core/testqgsexpression.cpp

```diff
@@ -2015,6 +2022,51 @@ class TestQgsExpression: public QObject
       QTest::newRow( "group by with null value" ) << "sum(\"col1\", \"col4\")" << false << QVariant( 8 );
     }
 
+    void maptip_display_data()
+    {
+      QTest::addColumn<QString>( "string" );
+      QTest::addColumn<QgsFeature>( "feature" );
+      QTest::addColumn<QgsVectorLayer *>( "layer" );
+      QTest::addColumn<bool>( "evalError" );
+      QTest::addColumn<QVariant>( "result" );
+
+      QgsFeature firstFeature = mPointsLayer->getFeature( 1 );
+      QgsVectorLayer *noLayer = nullptr;
+
+      QTest::newRow( "display not evaluated" ) << QStringLiteral( "display(@layer_id, $currentfeature, False)" ) << firstFeature << mPointsLayer  << false << QVariant( "Display expression with class = ' || Class" );
+      QTest::newRow( "display wrong layer" ) << QStringLiteral( "display()" ) << firstFeature << noLayer  << true << QVariant();
+      QTest::newRow( "display wrong feature" ) << QStringLiteral( "display()" ) << QgsFeature() << mPointsLayer << true << QVariant();
+      QTest::newRow( "maptip wrong feature" ) << QStringLiteral( "maptip()" ) << QgsFeature() << mPointsLayer << true << QVariant();
+      QTest::newRow( "maptip wrong layer" ) << QStringLiteral( "maptip()" ) << firstFeature << noLayer << true << QVariant();
+      QTest::newRow( "maptip not evaluated" ) << QStringLiteral( "maptip(@layer_id, $currentfeature, False)" ) << firstFeature << mPointsLayer << false << QVariant( "Maptip with class = [% \"Class\" %]" );
+
+      QTest::newRow( "display with 2 params" ) << QStringLiteral( "display(@layer_id, $currentfeature)") << firstFeature << mPointsLayer << false << QVariant( "Display expression with class = ' || Class" );
+      // QTest::newRow( "display without params" ) << QStringLiteral( "display()" ) << firstFeature << mPointsLayer << false << QVariant( "Display expression with class = Biplane" );
+      // QTest::newRow( "maptip without params" ) << QStringLiteral( "maptip()" ) << false << QVariant( "Maptip with class = Biplane" );
+      //QTest::newRow( "maptip" ) << QStringLiteral( "maptip('points', get_feature_by_id('points', '1'))" ) << false << QVariant( "Maptip with class = Biplane" );
```

Comment:
what's with this?

Reply:
No, I must enable these tests.
But it seems my tests is not working well, travis fails.
These new expressions are working in QGIS Desktop. But all these commented test and the test L2043 failed. The evaluation context seems wrong. Do you have a quick idea where the error can be.

Reply:
I think you'll need to add the layer to a project, and then add the project scope to your context

Reply:
Thanks, it should be OK now